src/boardToJSON.ts

```diff
@@ -0,0 +1,20 @@
+import {GameState} from "./gameState"
+
+export class BoardToJSON{
+    public static convert(board:number[][]):{"ShipPositions":Array<any>,"MyShots": Array<any>,"OpponentsShots": Array<any>} {
+        let ans:{"ShipPositions":Array<any>, "MyShots":Array<any>, "OpponentsShots":Array<any>} = {"ShipPositions":[],"MyShots":[], "OpponentsShots":[]  };
```

Comment:
Copying a comment across from the review for David & James' busboard:

So, this is an interesting one.

Basically, *every object is a JSON object in javascript* (after all, it stands for "JavaScript Object Notation"). Now, Typescript transpiles code into javascript (that's why you sett the .js & .js.map files), which means that, if we create the following in typescript:

```
export class TestObj {
    field: string;
    index: number;

    // some sensible constructor
}
```
and then instantiate it:
```
let testObj = new TestObj("tim", "1")
```
this will be seen by Javascript as:
```
{ field: "tim", index: 1 }
```

So while this is a really neat way of putting together JSON in any other language, in Typescript, you can simply create a new object to contain your stopPromise details, and return that!

(Please ask me if you have any follow up questions on this one!)